scripts/validate_min_versions_in_sync.py

```diff
@@ -41,6 +51,127 @@
+
+
+def get_operator_from(dependency) -> str:
+    if "<=" in dependency:
+        operator = "<="
+    elif ">=" in dependency:
+        operator = ">="
+    elif "=" in dependency:
+        operator = "="
+    elif ">" in dependency:
+        operator = ">"
+    elif "<" in dependency:
+        operator = "<"
+    else:
+        operator = None
+    return operator
+
+
+def get_yaml_map_from(yaml_dic) -> dict[str, str]:
+    yaml_map = {}
+    for dependency in yaml_dic:
+        if type(dependency) == dict or dependency in EXCLUSION_LIST:
+            continue
+        search_text = str(dependency)
+        operator = get_operator_from(search_text)
+        if "," in dependency:
+            yaml_dependency, yaml_version1 = search_text.split(",")
+            operator = get_operator_from(yaml_dependency)
+            yaml_package, yaml_version2 = yaml_dependency.split(operator)
+            yaml_version2 = operator + yaml_version2
+            yaml_map[yaml_package] = [yaml_version1, yaml_version2]
+        elif operator is not None:
+            yaml_package, yaml_version = search_text.split(operator)
+            yaml_version = operator + yaml_version
+            yaml_map[yaml_package] = [yaml_version]
+        else:
+            yaml_package, yaml_version = search_text.strip(), None
+            yaml_map[yaml_package] = yaml_version
+    return yaml_map
+
+
+def clean_version_list(yaml_versions, toml_version):
+    for i in range(len(yaml_versions)):
+        yaml_version = yaml_versions[i]
+        operator = get_operator_from(yaml_version)
+        if "<=" in operator or ">=" in operator:
+            yaml_version = yaml_version[2:]
+        else:
+            yaml_version = yaml_version[1:]
+        yaml_version = version.parse(yaml_version)
+        if yaml_version < toml_version:
+            yaml_versions[i] = "-" + str(yaml_version)
+        elif yaml_version >= toml_version:
+            if ">" in operator:
+                yaml_versions[i] = "-" + str(yaml_version)
+    return yaml_versions
+
+
+def pin_min_versions_to_yaml_file(yaml_map, toml_map, yaml_file_data):
+    data = yaml_file_data
+    for yaml_package, yaml_versions in yaml_map.items():
+        if yaml_package in EXCLUSION_LIST:
+            continue
+        old_dep = yaml_package
+        if yaml_versions is not None:
+            for yaml_version in yaml_versions:
+                old_dep += yaml_version + ", "
+            old_dep = old_dep[:-2]
+        if yaml_package in toml_map:
+            cleaned_yaml_versions = []
+            min_dep = toml_map[yaml_package]
+            if yaml_versions is None:
+                new_dep = old_dep + ">=" + min_dep
+                data = data.replace(old_dep, new_dep, 1)
+                continue
+            toml_version = version.parse(min_dep)
+            yaml_versions = clean_version_list(yaml_versions, toml_version)
+            [cleaned_yaml_versions.append(x) for x in yaml_versions if "-" not in x]
```

Comment:
same comment as above - can we create this in one go instead of first creating an empty list and then appending to it within a list comprehension?